webportal/src/app/cluster-view/hardware/hardware.component.js

```diff
@@ -17,14 +17,155 @@
+const loading = require('../../job/loading/loading.component');
 const webportalConfig = require('../../config/webportal.config.json');
 
 const hardwareHtml = hardwareComponent({
+  breadcrumb: breadcrumbComponent,
   grafanaUri: webportalConfig.grafanaUri
 });
 
-$("#sidebar-menu--cluster-view").addClass("active");
-$("#sidebar-menu--cluster-view--hardware").addClass("active");
+/*
+function createCORSRequest(method, url) {
+  var xhr = new XMLHttpRequest();
+  if ("withCredentials" in xhr) {
+    // Check if the XMLHttpRequest object has a "withCredentials" property.
+    // "withCredentials" only exists on XMLHTTPRequest2 objects.
+    xhr.open(method, url, true);
+  } else if (typeof XDomainRequest != "undefined") {
+    // Otherwise, check if XDomainRequest.
+    // XDomainRequest only exists in IE, and is IE's way of making CORS requests.
+    xhr = new XDomainRequest();
+    xhr.open(method, url);
+  } else {
+    // Otherwise, CORS is not supported by the browser.
+    xhr = null;
+  }
+  return xhr;
+}
+*/
 
-$('#content-wrapper').html(hardwareHtml);
\ No newline at end of file
+const loadMachines = () => {
+  /*
+  var xhr = createCORSRequest('GET', 'http://10.151.40.179:9090/api/v1/node');
+  if (!xhr) {
+    throw new Error('CORS not supported');
+  }
+  xhr.onload = function() {
+    var responseText = xhr.responseText;
+    alert(responseText);
+    console.log(responseText);
+    // process the response.
+  };
+  xhr.onerror = function() {
+    alert("ah!");
+    console.log('There was an error!');
+  };
+  xhr.send();
+  */
+  /*
+  $.ajax({
+
+    // The 'type' property sets the HTTP method.
+    // A value of 'PUT' or 'DELETE' will trigger a preflight request.
+    type: 'GET',
+  
+    // The URL to make the request to.
+    //url: 'http://html5rocks-cors.s3-website-us-east-1.amazonaws.com/index.html',
+    url: "http://10.151.40.179:9090/api/v1/node",
+  
+    xhrFields: {
+      // The 'xhrFields' property sets additional fields on the XMLHttpRequest.
+      // This can be used to set the 'withCredentials' property.
+      // Set the value to 'true' if you'd like to pass cookies to the server.
+      // If this is enabled, your server must respond with the header
+      // 'Access-Control-Allow-Credentials: true'.
+      withCredentials: true
+    },
+  
+    headers: {
+      // Set any custom headers here.
+      // If you set any non-simple headers, your server must include these
+      // headers in the 'Access-Control-Allow-Headers' response header.
+    },
+  
+    success: function(data) {
+      // Here's where you handle a successful response.
+      alert(data);
+    },
+  
+    error: function() {
+      // Here's where you handle an error response.
+      // Note that if the error was due to a CORS issue,
+      // this function will still fire, but there won't be any additional
+      // information about the error.
+    }
+  });
+  */
+  $.ajax({
+    url: "http://10.151.40.179:9090/api/v1/node",
```

Comment:
Once the NGinx proxy is ready, we can use the relative address thus removes hard coded host name and port right?

Reply:
@hwuu webportal config those URIs in env var and handle env in [webportal.config.js](https://github.com/Microsoft/pai/blob/master/webportal/config/webportal.config.js#L26-L30), pls keep this approach and do not hard code URI.

@sterowang In this case, it is easy for us to change the previous URI to proxy's prefix URI or relative path.

Reply:
Thanks guys. I have removed the hard-coded urls.